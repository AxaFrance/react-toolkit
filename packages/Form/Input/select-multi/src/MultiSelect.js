import React from 'react';
import PropTypes from 'prop-types';
import ReactSelect from 'react-select';

import {
  Input,
  InputConstants as Constants,
  withInput,
  omit,
} from '@axa-fr/react-toolkit-form-core';

const omitProperties = omit(['classModifier', 'className', 'isVisible']);

const MultiSelect = props => {
  const {
    name,
    loadOptions,
    values,
    options,
    value,
    onChange,
    onBlur,
    placeholder,
    className,
    readOnly,
    disabled,
    loadingPlaceholder,
    searchPromptText,
    noResultsText,
    ...otherProps
  } = props;
  
  const isDisabled = disabled || readOnly;
  const SelectComponent = loadOptions ? ReactSelect.Async : ReactSelect;

  if (values != null) {
    const newValues = options.reduce((acc, opt) => {
      const valueTemp = values.find(v => v === opt.value);
      if (valueTemp) {
        acc.push(opt);
      }
      return acc;
    }, []);

    return (
      <SelectComponent
        isMulti
        name={name}
        value={newValues}
        onChange={onChange}
        onBlur={onBlur}
        placeholder={placeholder}
        className={className}
        isDisabled={isDisabled}
        options={options}
        loadOptions={loadOptions}
        loadingPlaceholder={loadingPlaceholder}
        searchPromptText={searchPromptText}
        noResultsText={noResultsText}
        valueKey="value"
        labelKey="label"
        {...omitProperties(otherProps)}
      />
    );
  }
  const newValue = options.find(v => v.value === value) || '';
  return (
    <SelectComponent
      multi={false}
      name={name}
      value={newValue}
      placeholder={placeholder}
      onChange={onChange}
      onBlur={onBlur}
      className={className}
      isDisabled={isDisabled}
      options={options}
      loadOptions={loadOptions}
      loadingPlaceholder={loadingPlaceholder}
      searchPromptText={searchPromptText}
      noResultsText={noResultsText}
      valueKey="value"
      labelKey="label"
      {...omitProperties(otherProps)}
    />
  );
};

const propTypes = {
  ...Constants.propTypes,
  values: PropTypes.array,
  options: PropTypes.array.isRequired,
  placeholder: PropTypes.string,
  loadingPlaceholder: PropTypes.string,
  searchPromptText: PropTypes.string,
  noResultsText: PropTypes.string,
  loadOptions: PropTypes.func,
};

const defaultClassName = 'react-select';
const defaultProps = {
  ...Constants.defaultProps,
  values: null,
  placeholder: 'Select',
  loadingPlaceholder: 'Chargement...',
  searchPromptText: 'Saisir pour chercher',
  noResultsText: 'Aucun rÃ©sultat',
  loadOptions: null,
  className: defaultClassName,
};

const handlers = {
  onChange: ({ values, name, id, onChange }) => newValue => {
    if (values != null) {
      const newValues = newValue.map(v => v.value);
      onChange({
        values: newValues,
        name,
        id,
      });
    } else {
      onChange({
        value: newValue.value,
        name,
        id,
      });
    }
  },
};

const EnhancedComponent = withInput(
  defaultClassName,
  propTypes,
  defaultProps,
  handlers
)(MultiSelect);

EnhancedComponent.Clone = Input.Clone;
EnhancedComponent.displayName = MultiSelect.name;

export default EnhancedComponent;
