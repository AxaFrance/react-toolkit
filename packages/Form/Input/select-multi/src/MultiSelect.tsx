import React from 'react';
import ReactSelectAsync from 'react-select/async';
import { CustomFormEventValues } from '@axa-fr/react-toolkit-form-core';
import {
  OptionsType,
  OptionTypeBase,
  NamedProps as LibProps,
  ValueType,
} from 'react-select';

type Props = Omit<
  LibProps<OptionTypeBase>,
  'onChange' | 'loadOptions' | 'value'
> & {
  readOnly?: boolean;
  disabled?: boolean;
  onChange?: (event: CustomFormEventValues) => void;
  values?: string[];
  loadOptions?: (
    inputValue: string,
    callback: (options: OptionsType<OptionTypeBase>) => void
  ) => Promise<any> | void;
};

const MultiSelect = ({
  id,
  name,
  loadOptions,
  values,
  options,
  onChange,
  readOnly,
  disabled,
  placeholder = 'Select',
  className = 'react-select',
  noOptionsMessage = () => 'Aucun rÃ©sultat',
  ...otherProps
}: Props) => {
  const selectedValues = options.filter((opt) => values?.includes(opt.value));

  const loadOptionsDefault = () => Promise.resolve(options);

  const handleChange = (changedValue: ValueType<OptionTypeBase>) => {
    onChange &&
      onChange({
        id,
        name,
        values: (changedValue || []).map((v: OptionTypeBase) => v.value),
      });
  };
  return (
    <ReactSelectAsync
      {...otherProps}
      className={className}
      isMulti={values !== null}
      name={name}
      value={selectedValues}
      placeholder={placeholder}
      noOptionsMessage={noOptionsMessage}
      onChange={handleChange}
      isDisabled={disabled || readOnly}
      defaultOptions
      loadOptions={loadOptions || loadOptionsDefault}
      valueKey="value"
      labelKey="label"
    />
  );
};

export default MultiSelect;
