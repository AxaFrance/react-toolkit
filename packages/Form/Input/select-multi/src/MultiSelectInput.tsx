import React, { ComponentProps } from 'react';

import {
  Field,
  InputManager,
  FieldInput,
  HelpMessage,
} from '@axa-fr/react-toolkit-form-core';

import MultiSelect from './MultiSelect';

type Props = ComponentProps<typeof Field> &
  ComponentProps<typeof MultiSelect> & {
    helpMessage?: string;
  };

const MultiSelectInput = ({
  classModifier,
  message,
  children,
  helpMessage,
  id,
  disabled,
  classNameContainerLabel,
  classNameContainerInput,
  label,
  name,
  messageType,
  isVisible,
  forceDisplayMessage,
  className,
  ...multiSelectProps
}: Props) => {
  const inputFieldClassModifier = `${classModifier} ${
    disabled ? 'disabled' : ''
  }`;
  const inputId = InputManager.getInputId(id);
  return (
    <Field
      label={label}
      id={inputId}
      message={message}
      messageType={messageType}
      isVisible={isVisible}
      forceDisplayMessage={forceDisplayMessage}
      className={className}
      classModifier={classModifier}
      classNameContainerLabel={classNameContainerLabel}
      classNameContainerInput={classNameContainerInput}>
      <FieldInput
        className="af-form__select"
        classModifier={inputFieldClassModifier}>
        <MultiSelect disabled={disabled} {...multiSelectProps} />
        {children}
      </FieldInput>
      <HelpMessage message={helpMessage} isVisible={!message} />
    </Field>
  );
};

export default MultiSelectInput;
