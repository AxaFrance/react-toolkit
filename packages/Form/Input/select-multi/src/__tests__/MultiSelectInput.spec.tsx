import React from 'react';
import { render } from '@testing-library/react';
import MultiSelectInput from '../MultiSelectInput';
import { options, values } from './MultiSelect.spec';

test('renders MultiSelectInput correctly', () => {
  const { asFragment } = render(
    <MultiSelectInput
      label="Place type *"
      name="placeType"
      id="multiselectid"
      options={options}
      onChange={jest.fn()}
      values={values}
      helpMessage="Enter the place type"
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
