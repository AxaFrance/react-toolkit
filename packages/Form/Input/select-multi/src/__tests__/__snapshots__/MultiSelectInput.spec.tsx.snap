// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MultiSelect correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
  box-sizing: border-box;
}

.emotion-1 {
  z-index: 9999;
  border: 0;
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  width: 1px;
  position: absolute;
  overflow: hidden;
  padding: 0;
  white-space: nowrap;
}

.emotion-3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: hsl(0, 0%, 100%);
  border-color: hsl(0, 0%, 80%);
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  cursor: default;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  min-height: 38px;
  outline: 0!important;
  position: relative;
  -webkit-transition: all 100ms;
  transition: all 100ms;
  box-sizing: border-box;
}

.emotion-3:hover {
  border-color: hsl(0, 0%, 70%);
}

.emotion-4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 2px 8px;
  -webkit-overflow-scrolling: touch;
  position: relative;
  overflow: hidden;
  box-sizing: border-box;
}

.emotion-5 {
  background-color: hsl(0, 0%, 90%);
  border-radius: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 2px;
  min-width: 0;
  box-sizing: border-box;
}

.emotion-6 {
  border-radius: 2px;
  color: hsl(0, 0%, 20%);
  font-size: 85%;
  overflow: hidden;
  padding: 3px;
  padding-left: 6px;
  text-overflow: ellipsis;
  white-space: nowrap;
  box-sizing: border-box;
}

.emotion-7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 4px;
  padding-right: 4px;
  box-sizing: border-box;
}

.emotion-7:hover {
  background-color: #FFBDAD;
  color: #DE350B;
}

.emotion-8 {
  display: inline-block;
  fill: currentColor;
  line-height: 1;
  stroke: currentColor;
  stroke-width: 0;
}

.emotion-13 {
  margin: 2px;
  padding-bottom: 2px;
  padding-top: 2px;
  visibility: visible;
  color: hsl(0, 0%, 20%);
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: inline-grid;
  grid-area: 1/1/2/3;
  grid-template-columns: 0 min-content;
  box-sizing: border-box;
}

.emotion-13:after {
  content: attr(data-value) " ";
  visibility: hidden;
  white-space: pre;
  grid-area: 1/2;
  font: inherit;
  min-width: 2px;
  border: 0;
  margin: 0;
  outline: 0;
  padding: 0;
}

.emotion-14 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  box-sizing: border-box;
}

.emotion-15 {
  color: hsl(0, 0%, 80%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 8px;
  -webkit-transition: color 150ms;
  transition: color 150ms;
  box-sizing: border-box;
}

.emotion-15:hover {
  color: hsl(0, 0%, 60%);
}

.emotion-17 {
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  background-color: hsl(0, 0%, 80%);
  margin-bottom: 8px;
  margin-top: 8px;
  width: 1px;
  box-sizing: border-box;
}

<div
    class=" emotion-0"
  >
    <span
      class="emotion-1"
      id="react-select-2-live-region"
    />
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="emotion-1"
    />
    <div
      class=" emotion-3"
    >
      <div
        class=" emotion-4"
      >
        <div
          class="emotion-5"
        >
          <div
            class="emotion-6"
          >
            For fun
          </div>
          <div
            aria-label="Remove For fun"
            class="emotion-7"
            role="button"
          >
            <svg
              aria-hidden="true"
              class="emotion-8"
              focusable="false"
              height="14"
              viewBox="0 0 20 20"
              width="14"
            >
              <path
                d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
              />
            </svg>
          </div>
        </div>
        <div
          class="emotion-5"
        >
          <div
            class="emotion-6"
          >
            For drink
          </div>
          <div
            aria-label="Remove For drink"
            class="emotion-7"
            role="button"
          >
            <svg
              aria-hidden="true"
              class="emotion-8"
              focusable="false"
              height="14"
              viewBox="0 0 20 20"
              width="14"
            >
              <path
                d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
              />
            </svg>
          </div>
        </div>
        <div
          class=" emotion-13"
          data-value=""
        >
          <input
            aria-autocomplete="list"
            aria-expanded="false"
            aria-haspopup="true"
            autocapitalize="none"
            autocomplete="off"
            autocorrect="off"
            class=""
            id="react-select-2-input"
            role="combobox"
            spellcheck="false"
            style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
            tabindex="0"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class=" emotion-14"
      >
        <div
          aria-hidden="true"
          class=" emotion-15"
        >
          <svg
            aria-hidden="true"
            class="emotion-8"
            focusable="false"
            height="20"
            viewBox="0 0 20 20"
            width="20"
          >
            <path
              d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
            />
          </svg>
        </div>
        <span
          class=" emotion-17"
        />
        <div
          aria-hidden="true"
          class=" emotion-15"
        >
          <svg
            aria-hidden="true"
            class="emotion-8"
            focusable="false"
            height="20"
            viewBox="0 0 20 20"
            width="20"
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div>
      <input
        name="placeName"
        type="hidden"
        value="fun"
      />
      <input
        name="placeName"
        type="hidden"
        value="drink"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders MultiSelectInput correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
  box-sizing: border-box;
}

.emotion-1 {
  z-index: 9999;
  border: 0;
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  width: 1px;
  position: absolute;
  overflow: hidden;
  padding: 0;
  white-space: nowrap;
}

.emotion-3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: hsl(0, 0%, 100%);
  border-color: hsl(0, 0%, 80%);
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  cursor: default;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  min-height: 38px;
  outline: 0!important;
  position: relative;
  -webkit-transition: all 100ms;
  transition: all 100ms;
  box-sizing: border-box;
}

.emotion-3:hover {
  border-color: hsl(0, 0%, 70%);
}

.emotion-4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 2px 8px;
  -webkit-overflow-scrolling: touch;
  position: relative;
  overflow: hidden;
  box-sizing: border-box;
}

.emotion-5 {
  background-color: hsl(0, 0%, 90%);
  border-radius: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 2px;
  min-width: 0;
  box-sizing: border-box;
}

.emotion-6 {
  border-radius: 2px;
  color: hsl(0, 0%, 20%);
  font-size: 85%;
  overflow: hidden;
  padding: 3px;
  padding-left: 6px;
  text-overflow: ellipsis;
  white-space: nowrap;
  box-sizing: border-box;
}

.emotion-7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 4px;
  padding-right: 4px;
  box-sizing: border-box;
}

.emotion-7:hover {
  background-color: #FFBDAD;
  color: #DE350B;
}

.emotion-8 {
  display: inline-block;
  fill: currentColor;
  line-height: 1;
  stroke: currentColor;
  stroke-width: 0;
}

.emotion-13 {
  margin: 2px;
  padding-bottom: 2px;
  padding-top: 2px;
  visibility: visible;
  color: hsl(0, 0%, 20%);
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: inline-grid;
  grid-area: 1/1/2/3;
  grid-template-columns: 0 min-content;
  box-sizing: border-box;
}

.emotion-13:after {
  content: attr(data-value) " ";
  visibility: hidden;
  white-space: pre;
  grid-area: 1/2;
  font: inherit;
  min-width: 2px;
  border: 0;
  margin: 0;
  outline: 0;
  padding: 0;
}

.emotion-14 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  box-sizing: border-box;
}

.emotion-15 {
  color: hsl(0, 0%, 80%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 8px;
  -webkit-transition: color 150ms;
  transition: color 150ms;
  box-sizing: border-box;
}

.emotion-15:hover {
  color: hsl(0, 0%, 60%);
}

.emotion-17 {
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  background-color: hsl(0, 0%, 80%);
  margin-bottom: 8px;
  margin-top: 8px;
  width: 1px;
  box-sizing: border-box;
}

<div
    class="row af-form__group"
  >
    <div
      class="col-md-2"
    >
      <label
        aria-label="Place type *"
        class="af-form__group-label"
        for="multiselectid"
      >
        Place type *
      </label>
    </div>
    <div
      class="col-md-10"
    >
      <div
        class="af-form__select"
      >
        <div
          class=" emotion-0"
          id="multiselectid"
        >
          <span
            class="emotion-1"
            id="react-select-3-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="emotion-1"
          />
          <div
            class=" emotion-3"
          >
            <div
              class=" emotion-4"
            >
              <div
                class="emotion-5"
              >
                <div
                  class="emotion-6"
                >
                  For fun
                </div>
                <div
                  aria-label="Remove For fun"
                  class="emotion-7"
                  role="button"
                >
                  <svg
                    aria-hidden="true"
                    class="emotion-8"
                    focusable="false"
                    height="14"
                    viewBox="0 0 20 20"
                    width="14"
                  >
                    <path
                      d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="emotion-5"
              >
                <div
                  class="emotion-6"
                >
                  For drink
                </div>
                <div
                  aria-label="Remove For drink"
                  class="emotion-7"
                  role="button"
                >
                  <svg
                    aria-hidden="true"
                    class="emotion-8"
                    focusable="false"
                    height="14"
                    viewBox="0 0 20 20"
                    width="14"
                  >
                    <path
                      d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class=" emotion-13"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class=""
                  id="react-select-3-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class=" emotion-14"
            >
              <div
                aria-hidden="true"
                class=" emotion-15"
              >
                <svg
                  aria-hidden="true"
                  class="emotion-8"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                  />
                </svg>
              </div>
              <span
                class=" emotion-17"
              />
              <div
                aria-hidden="true"
                class=" emotion-15"
              >
                <svg
                  aria-hidden="true"
                  class="emotion-8"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div>
            <input
              name="placeType"
              type="hidden"
              value="fun"
            />
            <input
              name="placeType"
              type="hidden"
              value="drink"
            />
          </div>
        </div>
      </div>
      <small
        class="af-form__help"
      >
        Enter the place type
      </small>
    </div>
  </div>
</DocumentFragment>
`;
